src/value.c

```diff
@@ -224,6 +264,26 @@ SIValue SIValue_Divide(const SIValue a, const SIValue b) {
 	return SI_DoubleVal(SI_GET_NUMERIC(a) / (double)SI_GET_NUMERIC(b));
 }
 
+int SIArray_Compare(SIValue *arrayA, SIValue *arrayB) {
+
+	// check for the common range of indices
+	size_t arrayALen = array_len(arrayA);
+	size_t arrayBLen = array_len(arrayB);
```

Comment:
I would expect that if the arrays are of unequal length, we already know that they do not match and can return early (unless you've seen documentation that indicates otherwise).

Reply:
we should allow arrays to be sorted

Reply:
Agreed, but in the scenario:
```
create (n {arr1:[1]}), (m {arr1:[0,1]})
match (n) return n order by n.arr1
```
I'm guessing that the longer array is considered greater-than the shorter one, regardless of their contents.

The Cypher Query Language Reference v9 pdf has a section on comparability, orderability, and equivalence that hopefully resolves these questions!